controllers/storagecluster/storagecluster_controller.go

```diff
@@ -188,6 +189,36 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	storageConsumerPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil {
+				return true
+			}
+			old := e.ObjectOld.(*ocsv1alpha1.StorageConsumer)
+			new := e.ObjectNew.(*ocsv1alpha1.StorageConsumer)
+			return old.Status.Client.OperatorVersion != new.Status.Client.OperatorVersion
+		},
```

Comment:
Why only these 2 functions? we need to listen to all the others to remove the condition

Reply:
- the default value of event functions is `true` so we'll indeed get event for other conditions
- based on earlier comments I need to handle remove condition as well now, will do that